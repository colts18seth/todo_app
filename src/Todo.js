import React from 'react';

function Todo({ todo, remove }) {
    return (
        <li className="Todo">
            {todo}
            <button onClick={remove}>X</button>
        </li>
    );
}

export default Todo;