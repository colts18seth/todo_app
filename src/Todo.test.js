import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

test('renders without crashing', () => {
    render(<Todo />);
});

test('matches snapshot', () => {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

test("runs the delete function on button click", function () {
    const remove = jest.fn();
    const { getByText } = render(<Todo remove={remove} />);
    const removeButton = getByText("X");
    fireEvent.click(removeButton);
    expect(remove).toHaveBeenCalled();
});