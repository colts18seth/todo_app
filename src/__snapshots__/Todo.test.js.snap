// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="Todo"
  >
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;
